core/src/main/java/io/questdb/cutlass/http/processors/StaticContentProcessor.java

```diff
@@ -93,29 +101,37 @@ public LogRecord logInfoWithFd(HttpConnectionContext context) {
     @Override
     public void onRequestComplete(HttpConnectionContext context) throws PeerDisconnectedException, PeerIsSlowToReadException {
         final HttpRequestHeader headers = context.getRequestHeader();
-        Utf8Sequence url = headers.getUrl();
+        final Utf8Sequence url = headers.getUrl();
         logInfoWithFd(context).$("incoming [url=").$(url).$(']').$();
+
         if (Utf8s.containsAscii(url, "..")) {
             logInfoWithFd(context).$("URL abuse: ").$(url).$();
-            sendStatusTextContent(context, 404);
-        } else {
-            PrefixedPath path = prefixedPath.rewind();
-
-            if (url.size() == 1 && url.byteAt(0) == '/') {
-                path.concat(indexFileName);
-            } else {
-                path.concat(url);
-            }
+            sendStatusTextContent(context, HTTP_NOT_FOUND);
+            return;
+        }
 
-            LPSZ lpsz = path.$();
+        // hit redirects first, they can be outside the context path for the static files
+        final Utf8SequenceObjHashMap<Utf8Sequence> redirectMap = configuration.getRedirectMap();
+        int index = redirectMap.keyIndex(url);
+        if (index < 0) {
+            context.simpleResponse().sendStatusNoContent(HTTP_MOVED_PERM, redirectMap.valueAt(index));
```

Comment:
when redirecting we need to copy all URL params to the header, otherwise we lose them, and that breaks web console SOO